config_defaults_inc.php

```diff
@@ -2038,60 +2038,89 @@
 
 
 /**
- * Specifies the LDAP or Active Directory server to connect to.
+ * Specifies the LDAP or Active Directory server(s) to connect to.
+ * Multiple servers can be specified as a space-separated list.
  *
  * This must be a full LDAP URI (ldap[s]://hostname:port)
- * - Protocol can be either ldap or ldaps (for SSL encryption). If omitted,
- *   then an unencrypted connection will be established on port 389.
+ * - Protocol must be either:
+ *   - ldap (unencrypted or opportunistic StartTLS)
+ *   - ldaps (for TLS encryption)
  * - Port number is optional, and defaults to 389. If this doesn't work, try
  *   using one of the following standard port numbers: 636 (ldaps); for Active
  *   Directory Global Catalog forest-wide search, use 3268 (ldap) or 3269 (ldaps)
  *
  * Examples of valid URI:
- *   ldap.example.com
  *   ldap://ldap.example.com
  *   ldaps://ldap.example.com:3269/
  *
  * @global string $g_ldap_server
  */
-$g_ldap_server = 'ldaps://ldap.example.com/';
+$g_ldap_server = 'ldap://ldap.example.com/';
 
 /**
- * The root distinguished name for LDAP searches
+ * Decides if the connection will attempt an opportunistic upgrade to
+ * a TLS connection (STARTTLS).
+ * Be sure to use the ldap:// scheme and not ldaps://
+ * if setting this to ON.  For security, a failure aborts the entire connection.
+ *
+ * @global bool $g_ldap_use_starttls
```

Comment:
After latest changes, change to `@global integer $g_ldap_use_starttls`

Reply:
Why should it be an integer?  A use / don't use choice is by definition a boolean.

Reply:
Booleans can be either true or false. https://www.php.net/manual/en/language.types.boolean.php
You set it to constant `ON` which is defined as `1`.

Reply:
I set it to `ON` because you told me to :)

As I said, I don't know PHP, I figured ON/OFF where synonymous with true/false.

Why did you suggest using ON instead of true?  Why is integer better than boolean here?

Reply:
> I set it to `ON` because you told me to :)

Wasn't me, was @vboctor

> Why is integer better than boolean here?

Not sure what @vboctor intended when requesting the change.

Maybe because you could add more options later on without that much friction.
Maybe because we use integers most of the time for such kind of options (e.g. `$g_enable_email_notification`)

Furthermore integers are easier to handle than booleans at various places, e.g, when persisting them in different database systems and/or different languags.

Of course, we are not complete clean, as there is `$g_reassign_on_feedback` which is currently set to `ON` but also documented as boolean.
And there is `$g_session_save_path` which is currently set to `false`,  documented as boolean, but can also be set to a string.

> As I said, I don't know PHP

so you are certainly confused by the dynamic type system of PHP (still the same for me, even after some years of PHP)

Reply:
>Wasn't me, was @vboctor

Sorry, I meant the plural 'you' (darn English).

OK, I defer to 'yous' on this...